pkg/configuration/configuration.go

```diff
@@ -0,0 +1,208 @@
+// Package configuration is in charge of the validation and extraction of all
+// the configuration details from a configuration file or environment variables.
+package configuration
+
+import (
+	"strings"
+	"time"
+
+	errs "github.com/pkg/errors"
+	"github.com/spf13/viper"
+)
+
+const (
+	// EnvPrefix will be used for environment variable name prefixing
+	EnvPrefix = "REGISTRATION"
+
+	// VersionKey is the key for the version config value
+	VersionKey = "version"
+
+	// Constants for viper variable names. Will be used to set
+	// default values as well as to get each value
+	varHTTPAddress = "http.address"
+	// DefaultHTTPAddress is the address and port string that your service will
+	// be exported to by default.
+	DefaultHTTPAddress = "0.0.0.0:8080"
+
+	varHTTPIdleTimeout = "http.idle_timeout"
+	// DefaultHTTPIdleTimeout specifies the default timeout for HTTP idling
+	DefaultHTTPIdleTimeout = time.Second * 15
+
+	varHTTPCompressResponses = "http.compress"
+	// DefaultHTTPCompressResponses compresses HTTP responses for clients that
+	// support it via the 'Accept-Encoding' header.
+	DefaultHTTPCompressResponses = false
+
+	varLogLevel = "log.level"
+	// DefaultLogLevel is the default log level used in your service.
+	DefaultLogLevel = "info"
+
+	varLogJSON = "log.json"
+	// DefaultLogJSON is a switch to toggle on and off JSON log output.
+	DefaultLogJSON = false
+
+	varGracefulTimeout = "graceful_timeout"
+	// DefaultGracefulTimeout is the duration for which the server gracefully
+	// wait for existing connections to finish - e.g. 15s or 1m
+	DefaultGracefulTimeout = time.Second * 15
+
+	varHTTPWriteTimeout = "http.write_timeout"
+	// DefaultHTTPWriteTimeout specifies the default timeout for HTTP writes
+	DefaultHTTPWriteTimeout = time.Second * 15
+
+	varHTTPReadTimeout = "http.read_timeout"
+	// DefaultHTTPReadTimeout specifies the default timeout for HTTP reads
+	DefaultHTTPReadTimeout = time.Second * 15
+
+	varHTTPInsecure = "http.insecure"
```

Comment:
all these HTTPS/cert related variables are not needed (will run in HTTP inside the container, secured HTTPS connection will be handled at the cluster's edge router)

Reply:
Yes, but the SSL support is only a few lines (without the configuration options, it's actually <5 lines), and as this is more or less a scaffolding, I would actually leave it in. Maybe we need it when we're possibly re-using this scaffolding for other services.

Reply:
I would do that in the non-scaffolding part to keep the base open for any use. But to avoid more discussion here, I'll remove it.